server/api/utils/builder.py

```diff
@@ -400,6 +391,14 @@ def build_image(
     runtime_spec = runtime.spec if runtime else None
     runtime_builder_env = runtime_spec.build.builder_env or {}
 
+    project_default_function_node_selector = {}
+    if runtime and runtime._get_db():
+        project_obj = runtime._get_db().get_project(runtime.metadata.project)
+        if project_obj:
+            project_default_function_node_selector = (
+                project_obj.spec.default_function_node_selector
+            )
+
```

Comment:
Lets move this to a util like `resolve_project_default_service_account` then you don't need to add this parameter everywhere

Reply:
We decided not to do this because then we'll need to pass the runtime object parameter everywhere